### PR TITLE
Add DcpToInterchange class

### PR DESCRIPTION
**Pull request by zakn-amd**
For conversion of DCP into a Logical and Physical Interchange netlist, along with (normal) XDC constraints.

Also accessible through `rapidwright DcpToInterchange` due to addition to `MainEntrypoint`.

### CODE REVIEW

File: src/com/xilinx/rapidwright/interchange/DcpToInterchange.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        LogNetlistWriter.writeLogNetlist(design.getNetlist(), logNlistName);
+        PhysNetlistWriter.writePhysNetlist(design, physNlistName);
+
+        List<String> constraints = design.getXDCConstraints(ConstraintGroup.NORMAL);
+        try (FileOutputStream f = new FileOutputStream(xdcName)) {
+            XDCParser.writeXDC(constraints, f);
+        }
+    }
+}
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/interchange/DcpToInterchange.java:58]:
```suggestion
}

```

File: src/com/xilinx/rapidwright/interchange/DcpToInterchange.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if (args.length != 1) {
+            System.out.println("USAGE: <input.dcp>");
+            return;
+        }
+
+        Design design = Design.readCheckpoint(args[0]);
+        String baseName = Paths.get(args[0]).getFileName().toString();
+        baseName = FileTools.removeFileExtension(baseName);
+        String logNlistName = baseName + ".netlist";
+        String physNlistName = baseName + ".phys";
```
Comment (clavin-xlnx) [inline src/com/xilinx/rapidwright/interchange/DcpToInterchange.java:47]: I'm okay with `logName`/`physName` or `logNetlistName`/`physNetlistName`

File: src/com/xilinx/rapidwright/edif/EDIFTools.java [hunk 3]
```diff
@@ -842,7 +842,7 @@ public static EDIFNetlist loadEDIFFile(Path fileName) {
     public static EDIFNetlist loadEDIFFile(Path fileName, int maxThreads) {
         try {
             final long size = Files.size(fileName);
-            if (ParallelEDIFParser.calcThreads(size, maxThreads, fileName.toString().endsWith(".gz")) > 1) {
+            if (true /*ParallelEDIFParser.calcThreads(size, maxThreads, fileName.toString().endsWith(".gz")) > 1*/) {
```
Comment (clavin-xlnx) [inline src/com/xilinx/rapidwright/edif/EDIFTools.java:845]: Is this an intentional change?

Comment (eddieh-xlnx) [reply src/com/xilinx/rapidwright/edif/EDIFTools.java]: Good catch. Reverted!

**Review by clavin-xlnx — APPROVED** [review]

**Pull request merged**